chanzuckerberg/single-cell-data-portal
# feat: Update filter UI to match mocks (#4558)

## Reason for Change

- Collection and Datasets index filter, and Sidebar component ui updated to match mocks [here](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4626-43440&t=eneD3qBxvN1BSNdk-4) and [here](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4626-43446&t=eneD3qBxvN1BSNdk-4).
- #4558

## Changes

- Updated `Sidebar` component to match [mocks](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4626-41034&t=eneD3qBxvN1BSNdk-4).
  - Includes `Sidebar` vertical separator and toggle button.
  - Vertical separator updates are global - see [slack](https://clevercanary.slack.com/archives/C02CF7MQPGV/p1684256759227059?thread_ts=1684236150.259999&cid=C02CF7MQPGV).
  - Minor refactoring (removal) of unused `Sidebar` props.
- Updated `Filter` component(s) to match mocks [here](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4626-43440&t=eneD3qBxvN1BSNdk-4) and [here](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4626-43446&t=eneD3qBxvN1BSNdk-4).
  - Updates include filter menu button and associated chevron icon, filter menu popover, menu list item including icon, text and count, and selected filter tags.
  - For each of these elements, various modifications have been made, including adjustments to colours, fonts, and spacings.
  - Includes style refactoring to create a single source of truth for menu item styles.
  - Where possible, Blueprint components have been replaced with either SDS or Mui components.
  - Where possible, text styles, colours and spacings have referenced the Single Cell SDS theme file.

## Testing steps

Collections: https://cc-filter-ui-frontend.rdev.single-cell.czi.technology/collections
Datasets: https://cc-filter-ui-frontend.rdev.single-cell.czi.technology/datasets

- Collections and Datasets sidebar and filters should match mocks [here](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4626-43440&t=eneD3qBxvN1BSNdk-4) and [here](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?type=design&node-id=4626-43446&t=eneD3qBxvN1BSNdk-4).
- WMG sidebar should render new vertical separator.
- WMG sidebar toggle button font-weight and margin-bottom should remain unchanged.
